test/distributed/pipelining/schedule_registry.py

```diff
@@ -172,3 +172,50 @@ def __init__(
                 _Action(1, W, 1),
             ],
         }
+
+
+class ScheduleWithReorderedB(PipelineScheduleMulti):
+    n_stages = 4
+    num_microbatches = 2
+    rank_stages = {
+        0: [0, 2],
+        1: [1, 3],
+    }
+
+    def __init__(
+        self,
+        stages: List[_PipelineStageBase],
+        n_microbatches: int,
+        loss_fn: Optional[Callable] = None,
+    ):
+        super().__init__(
+            stages=stages,
+            n_microbatches=n_microbatches,
+            loss_fn=loss_fn,
+        )
+        # Go through two microbatches
+        self.pipeline_order = {
+            0: [
+                _Action(0, F, 0),
+                _Action(0, F, 1),
+                _Action(2, F, 0),
+                _Action(2, F, 1),
+                None,
+                None,
+                _Action(2, B, 0),
```

Comment:
i'm probably just missing something but isn't this schedule ordered in the usual order,  i.e. not reordered B?

the usual order would execute B's in increasing microbatch_id sequence right?

Reply:
Yeah good point, I realized the test is not good because the error would only surface if we use send for mb A after we did a backward for mb B, which the test is not catching. So i updated it to reorder the send operations in the new schedule